get_defs.py

```diff
@@ -5,40 +5,45 @@ def api_call(word: str) -> list:
     
     trial = 0
     time.sleep(0.5)  #0.4 sleep with 2 attempts has had best results
+
     while trial < 2:
-        trial +=1
+
         try:
             url = f"https://api.dictionaryapi.dev/api/v2/entries/en/{word}"
             response = requests.get(url)
 
             if response.status_code == 200:
                     data = response.json()
-                    #print(f"Data found for {word}")
-                    return data
-            
+                    return data          
             else:
-                print(f"This was the error code: {response.status_code}")
-        except:
+                trial +=1
+                raise Exception(f"non-200 status code returned for {word}: {response.status_code}")
+            
+
+        except Exception:
             print(f"Bad Request - no relvant url found for {word}")
             return None
     
 
 #Returns a set of all found vocab types for a given word (e.g. {"verb", "noun"})
 def word_types(word):
     
-    word_types = set({})
+    word_types = set()
     current_call = api_call(word)
     try:
         if current_call != None:
             for items in current_call:
                 for mean_items in items['meanings']:
                     word_types.add(mean_items['partOfSpeech'])
-            if len(list(word_types)) > 0:
+            if len((word_types)) > 0:
                 return word_types
-            else:
-                return None
-    except:
+
+    except IndexError:
+        print(f"Index Error raised in word_types for {word}.")
         return None
+    
+    else:
+        print(f"Non-index error raised in word_types for {word}.")
```

Comment:
There is a consistency discrepancy here.

As this is a returning function, it is the done thing to never not return. Failing to return implies that additional logic can safely be added after the statement, which isn't the case here as this is raising an error.

Additionally, `else` is called when no exception is raised in the try block (its not commonly used). I think you meant `except:` to handle every raised value that isn't an IndexError.

Reply:
This has now been altered to:

`def word_types(word):
    
    """Returns a set of all found vocab types for a given word from 
    api (e.g. {"verb", "noun"})
    """

    word_types = set()
    current_call = api_call(word)

    if current_call != False:
        for items in current_call:
            for mean_items in items['meanings']:
                word_types.add(mean_items['partOfSpeech'])
        if len((word_types)) > 0:
            return word_types

    else:
        print("Should never run.")`